restcomm.android.sdk/src/main/java/org/restcomm/android/sdk/fcm/FcmMessagingService.java

```diff
@@ -44,7 +44,7 @@ private void sendNotification() {
         //start service
         //we dont need to start a service if its already running.
         //The restcomm server will always send the push...
-        if (!RCDevice.isServiceAttached){
+        if (!RCDevice.isServiceAttached && !RCDevice.messageReceivingInProgress){
```

Comment:
I'd prefer to not expose yet another state variable from RCDevice as it increases coupling. Can we use isServiceInitialized instead?